src/beyond_identity/api/common/request.rs

```diff
@@ -0,0 +1,104 @@
+use super::api_client::ApiClient;
+use crate::common::error::BiError;
+use http::Method;
+use serde::{de::DeserializeOwned, Serialize};
+
+pub async fn send_request<T, U>(
```

Comment:
I think send_request and send_request_paginated should be implemented on the ApiClient:

```rust
impl ApiClient {
    pub async fn send_request<T, U>(
        &self,
        method: Method,
        url: &str,
        body: Option<&T>,
    ) -> Result<U, BiError>
    where
        T: Serialize,
        U: DeserializeOwned,
    { todo!() }

    pub async fn send_request_paginated<T, U>(
        &self,
        method: Method,
        url: &str,
        body: Option<&T>,
        items_key: &str,
    ) -> Result<Vec<U>, BiError>
    where
        T: Serialize,
        U: DeserializeOwned,
    { todo!() }
}
```

That way, we can call them as `api_client.send_request` which makes more sense than `send_request(api_client, ...)`